servo-build-dependencies/init.sls

```diff
@@ -36,16 +37,43 @@ servo-dependencies:
       - xpra
       - xserver-xorg-input-void
       - xserver-xorg-video-dummy
+      {% elif grains['os'] in ['CentOS', 'Fedora'] %}
+      - bzip2-devel
+      - cabextract
+      - curl
+      - dbus-devel
+      - expat-devel
+      - fontconfig-devel
+      - freeglut-devel
+      - freetype-devel
+      - gcc-c++
+      - glib2-devel
+      - gperf
+      - libtool
+      - libX11-devel
+      - libXcursor-devel
+      - libXi-devel
+      - libXmu-devel
+      - libXrandr-devel
+      - llvm-devel
+      - mesa-libEGL-devel
+      - mesa-libGL-devel
+      - mesa-libOSMesa-devel
+      - openssl-devel
+      - rpm-build
+      - ttmkfdir
       {% endif %}
+  {% if salt['pillar.get']('fully_managed', True) %}
   pip.installed:
     - pkgs:
       - ghp-import
       - s3cmd
     - require:
       - pkg: pip
       - pip: virtualenv
+  {% endif %}
 
-{% if grains['os'] == 'Ubuntu' %}
+{% if grains['os'] == 'Ubuntu' and grains['oscodename'] == 'trusty' %}
 multiverse:
   pkgrepo.managed:
     - name: 'deb http://archive.ubuntu.com/ubuntu trusty multiverse'
```

Comment:
Instead of gating on being Trusty, try replacing `trusty` on this line with `{{ grains['oscodename'] }}`. I'm not 100% sure but I think it should work on Trusty and Xenial both.